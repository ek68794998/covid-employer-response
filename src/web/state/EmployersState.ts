import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}