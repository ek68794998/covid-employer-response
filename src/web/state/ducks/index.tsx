import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});