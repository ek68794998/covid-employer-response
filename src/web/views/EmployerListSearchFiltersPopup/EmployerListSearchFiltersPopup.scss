@import "../Colors.scss";

.EmployerListSearchFiltersPopup__Container {
	background: $background;
	border-radius: 8px;
	box-shadow: 0 2px 5px $box-shadow-color;
	display: flex;
	flex-direction: column;
	margin-top: 4px;
	padding: 8px;
	position: absolute;
	width: 200px;
	z-index: 25;

	h3 {
		font-size: 18px;
		margin: 0;
		padding: 4px;
	}
}

.EmployerListSearchFiltersPopup__Overlay {
	bottom: 0;
	left: 0;
	position: fixed;
	right: 0;
	top: 0;
	z-index: 24;
}
