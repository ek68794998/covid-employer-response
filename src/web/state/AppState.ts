import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}