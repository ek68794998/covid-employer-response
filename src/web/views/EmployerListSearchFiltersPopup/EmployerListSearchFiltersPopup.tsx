import React from "react";
import { useSelector } from "react-redux";
import { RouteProps } from "react-router-dom";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { getStrings } from "../../state/ducks/localization/selectors";

import EmployerListSearchFiltersSelector from "../EmployerListSearchFiltersSelector/EmployerListSearchFiltersSelector";
import "./EmployerListSearchFiltersPopup.scss";

export interface SelectorProps {
	initialValue: boolean;

	label: string;

	onChange: React.ChangeEventHandler<HTMLInputElement>;
}

interface Props extends RouteProps {
	childProps: SelectorProps[];

	multiselect: boolean;

	onClose: () => void;

	title: string;
}

const EmployerListSearchFiltersPopup: React.FC<Props> = (props: Props): React.ReactElement => {
	const { childProps, multiselect, onClose, title } = props;

	const children: JSX.Element[] = childProps.map((p: SelectorProps, i: number) => (
		<EmployerListSearchFiltersSelector
			key={i}
			initialValue={p.initialValue}
			label={p.label}
			multiselect={multiselect}
			onChange={p.onChange}
		/>
	));

	return (
		<>
			<div className="EmployerListSearchFiltersPopup__Container">
				<h3>{title}</h3>
				{children}
			</div>
			<div className="EmployerListSearchFiltersPopup__Overlay" onClick={onClose} />
		</>
	);
};

export default EmployerListSearchFiltersPopup;
