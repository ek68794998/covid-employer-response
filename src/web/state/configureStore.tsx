import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;