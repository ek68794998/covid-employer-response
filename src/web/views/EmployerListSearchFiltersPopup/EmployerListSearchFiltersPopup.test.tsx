import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { mockComponent } from "../../../__tests__/TestUtils";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerListSearchFiltersPopup from "./EmployerListSearchFiltersPopup";

jest.mock(
	"../EmployerListSearchFiltersSelector/EmployerListSearchFiltersSelector",
	(): any => mockComponent("EmployerListSearchFiltersSelector"));

describe("<EmployerListSearchFiltersPopup />", (): void => {
	test("renders without exploding", (): void => {
		const store: Store<AppState, AnyAction> = configureStore({});

		const renderedValue: ReactTestRenderer =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListSearchFiltersPopup
							childProps={[]}
							multiselect={false}
							onClose={(): void => { /* Do nothing. */ }}
							title={"Title"}
						/>
					</BrowserRouter>
				</Provider>,
			);

		expect(renderedValue.toJSON()).toMatchSnapshot();
	});
});
