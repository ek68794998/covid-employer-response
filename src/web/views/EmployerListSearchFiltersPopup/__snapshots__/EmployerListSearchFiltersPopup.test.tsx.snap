// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerListSearchFiltersPopup /> renders without exploding 1`] = `
Array [
  <div
    className="EmployerListSearchFiltersPopup__Container"
  >
    <h3>
      locationDefault string
    </h3>
  </div>,
  <div
    className="EmployerListSearchFiltersPopup__Overlay"
    onClick={[Function]}
  />,
]
`;
